Document missing CAxxxx rules (#27885)
dotnet/docs: docs/fundamentals/code-analysis/quality-rules/ca2257.md

---

> title: "CA2257: Members defined on an interface with the 'DynamicInterfaceCastableImplementationAttribute' should be 'static'"

Reviewer: ```suggestion
title: "CA2257: Members defined on an interface with 'DynamicInterfaceCastableImplementationAttribute' should be 'static'"
```

> description: "Learn about code analysis rule CA2257: Members defined on an interface with the 'DynamicInterfaceCastableImplementationAttribute' should be 'static'"

Reviewer: ```suggestion
description: "Learn about code analysis rule CA2257: Members defined on an interface with 'DynamicInterfaceCastableImplementationAttribute' should be 'static'"
```

ms.date: 01/19/2022
ms.topic: reference
f1_keywords:
- CA2257
- DynamicInterfaceCastableImplementationAnalyzer
helpviewer_keywords:
- DynamicInterfaceCastableImplementationAnalyzer
- CA2257
author: Youssef1313
---
# CA2257: Members defined on an interface with the 'DynamicInterfaceCastableImplementationAttribute' should be 'static'

|                                     | Value                                |
| ----------------------------------- | ------------------------------------ |
| **Rule ID**                         | CA2257                               |
| **Category**                        | [Usage](usage-warnings.md)           |
| **Fix is breaking or non-breaking** | Non-breaking                         |

## Cause

> An interface member isn't explicitly implemented nor static.

Reviewer: ```suggestion
An interface member isn't explicitly implemented or marked `static`.
```

## Rule description

> Since a type that implements 'IDynamicInterfaceCastable' may not implement a dynamic interface in metadata, calls to an instance interface member that is not an explicit implementation defined on this type are likely to fail at runtime. Mark new interface members 'static' to avoid runtime errors.

Reviewer: ```suggestion
Since a type that implements `IDynamicInterfaceCastable` may not implement a dynamic interface in metadata, calls to an instance interface member that is not an explicit implementation defined on this type are likely to fail at run time. To avoid run-time errors, mark new interface members `static`.
```

## How to fix violations

> Make the interface member static.

Reviewer: ```suggestion
Mark the interface member `static`.
```
